test/signing_test.go

```diff
@@ -8,15 +8,420 @@ import (
 
 	"github.com/Azure/dalec"
 	"github.com/Azure/dalec/test/testenv"
+	"github.com/moby/buildkit/client/llb"
 	gwclient "github.com/moby/buildkit/frontend/gateway/client"
 	"github.com/stretchr/testify/assert"
 )
 
-func distroSigningTest(t *testing.T, spec *dalec.Spec, buildTarget string) testenv.TestFunc {
+func runTest(t *testing.T, f testenv.TestFunc) {
+	t.Helper()
+	ctx := startTestSpan(baseCtx, t)
+	testEnv.RunTest(ctx, t, f)
+}
+
+func newSpec() *dalec.Spec {
+	spec := fillMetadata("foo", &dalec.Spec{
+		Sources: map[string]dalec.Source{
+			"foo": {
+				Inline: &dalec.SourceInline{
+					File: &dalec.SourceInlineFile{
+						Contents: "#!/usr/bin/env bash\necho \"hello, world!\"\n",
+					},
+				},
+			},
+		},
+		Build: dalec.ArtifactBuild{
+			Steps: []dalec.BuildStep{
+				{
+					Command: "/bin/true",
+				},
+			},
+		},
+		Artifacts: dalec.Artifacts{
+			Binaries: map[string]dalec.ArtifactConfig{
+				"foo": {},
+			},
+		},
+	})
+
+	return spec
+}
+
+func linuxSigningTests(ctx context.Context, testConfig testLinuxConfig) func(*testing.T) {
+	return func(t *testing.T) {
+		newSpec := func() *dalec.Spec {
```

Comment:
This mostly looks like a dup of the actual `newSpec`?

Reply:
It set up a signing config. I refactored it to deduplicate.

Reply:
done